transformer_engine/pytorch/module/fp8_padding.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+"""FP8 Padding API"""
+
+from typing import Union, List
+
+import torch
+
+import transformer_engine_torch as tex
+
+from ..cpp_extensions import (
+    multi_padding_fused,
+)
+from ..jit import no_torch_dynamo
+
+
+__all__ = ["Fp8Padding"]
+
+
+class _Fp8Padding(torch.autograd.Function):
+    """functional FP8 padding"""
+
+    @staticmethod
+    def forward(
+        ctx,
+        inp: torch.Tensor,
+        m_splits: List[int],
+        padded_m_splits: List[int],
+        is_grad_enabled: bool,
+    ) -> torch.Tensor:
+        # Make sure input dimensions are compatible
+        in_features = inp.shape[-1]
+        inputmats = torch.split(inp.view(-1, in_features), m_splits)
+
+        # Allocate cast and transpose output tensor
+        total_row = sum(padded_m_splits)
+        out = torch.empty([total_row, in_features], dtype=inp.dtype, device="cuda")
+        out_list = torch.split(out, padded_m_splits)
+
+        multi_padding_fused(inputmats, padded_m_splits, out_list)
```

Comment:
I think it's better to make `multi_padding_fused` accept whole tensors as inp/out, 
```python
inp = inp.view(-1, in_features)
multi_padding_fused(inp, m_splits, padded_m_splits, out)
```
and replace `torch.split` with stepping the pointers in C++ to reduce CPU overheads, as I'm doing in https://github.com/NVIDIA/TransformerEngine/pull/1128/files#diff-342b0e9e5b472b443484b3a2c4a78647cd72431c272ed44c678ecbe636fb7a3aR191-R194

Reply:
Ack. Your work is helpful to me. 

Thx.